clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -1678,6 +1678,30 @@ half3 max(half3, half3);
 _HLSL_16BIT_AVAILABILITY(shadermodel, 6.2)
 _HLSL_BUILTIN_ALIAS(__builtin_elementwise_max)
 half4 max(half4, half4);
+_HLSL_16BIT_AVAILABILITY(shadermodel, 6.2)
+constexpr half2 max(half2 p0, half p1) {
```

Comment:
If a user were to call the max function with, say, a half2 and a half3, is it our intention to reject that / emit an error?

Reply:
We only want to allow things of the form max(vector<T,N>, T) or max(T, vector<T,N>)